docs/pages/api/rbac.mdx

```diff
@@ -0,0 +1,1017 @@
+    'create-teleport-role': 'true' 
+subjects:
+- kind: Group
+  name: app-developer
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role 
+  name: pod-reader 
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Create the resources: 
+
+```code
+$ kubectl apply -f pod-reader.yaml
+namespace/app created
+role.rbac.authorization.k8s.io/pod-reader created
+rolebinding.rbac.authorization.k8s.io/read-pods created
+```
+
+Next, define a cluster role and cluster role binding that allow users in the
+`ops` group to read, create, and execute commands on pods in all namespaces. Add
+the following to a file called `pod-ops.yaml`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pod-ops
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list", "create", "exec", "logs"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: pod-ops
+  annotations:
+    'create-teleport-role': 'true' 
+subjects:
+- kind: Group
+  name: ops
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: pod-ops
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Create the resources:
+
+```code
+$ kubectl apply -f pod-ops.yaml
+clusterrole.rbac.authorization.k8s.io/pod-ops created
+clusterrolebinding.rbac.authorization.k8s.io/pod-ops created
+```
+
+Later in this guide, we will show you how to automatically generate Teleport
+roles based on the Kubernetes RBAC resources you created.
+
+### Define RBAC resources for the client application
+
+Next, ensure that your API client can read the RBAC resources you created.
+Create a file called `rbac-sync.yaml` with the following content:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rbac-sync
+rules:
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources: ["roles", "clusterroles", "rolebindings", "clusterrolebindings"]
+  verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rbac-sync
+subjects:
+- kind: User
+  name: sync-kubernetes-rbac
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: rbac-sync
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply the changes:
+
+```code
+$ kubectl apply -f rbac-sync.yaml
+clusterrole.rbac.authorization.k8s.io/rbac-sync created
+clusterrolebinding.rbac.authorization.k8s.io/rbac-sync created
+```
+
+## Step 2/6. Set up Teleport 
+
+In this step, you will configure Teleport to enable your API client application
+to interact with your Kubernetes cluster.
+
+### Create a user and role for the client application
+
+Give the client application a Teleport user and role that can retrieve
+information about a Kubernetes cluster that is registered with Teleport,
+authenticate to the cluster, and create or update Teleport roles.
+
+Create a file called `sync-kubernetes-rbac.yaml` with the following content:
+
+```yaml
+kind: role
+version: v5
```

Comment:
Should we use v6 roles? Or just keep v5 to make backports easier?

Reply:
I've changed it to v6. I'll make a note to use v5 when we backport